### PR TITLE
Adding tests for all branches of OAuth2Credentials._do_revoke.

### PR DESCRIPTION
**Pull request by dhermes**
Also updating the `_do_refresh_request tests` to check the content sent.

### CODE REVIEW

File: tests/test_client.py [hunk 1]
```diff
@@ -1218,6 +1225,85 @@ def test__do_refresh_request_failure_w_json_error_and_desc(self):
...
+            'status': http_client.OK,
+        })
+        self._do_revoke_test_helper(response, b'', None)
+
+    def test__do_revoke_success_with_store(self):
+        response = httplib2.Response({
+            'status': http_client.OK,
+        })
+        store = mock.MagicMock()
+        self._do_revoke_test_helper(response, b'', None, store=store)
```
Comment (theacodes) [inline tests/test_client.py:1271]: no assert on the store being called?

**Comment by jonparrott** [comment]
Looks mostly good. Feel free to merge when you respond to or address my one comment.

File: tests/test_client.py (hunk 1 above)
Comment (dhermes) [inline tests/test_client.py:1271]: It happens in the helper.

Comment (theacodes) [inline tests/test_client.py:1271]: Cool. LGTM.

**Comment by nathanielmanistaatgoogle** [comment]
Looks great.

**Pull request merged**